src/configuration.ts

```diff
@@ -352,6 +405,15 @@ export class ConfigurationManager implements vscode.Disposable
 
     private async addVcpkgToolchain(vcpkgRoot : string)
     {
+        let originToolchain = this.getCMakeVcpkgToolchain();
+        if (originToolchain !== undefined)
```

Comment:
This can be a bit confusing:
1. What should we do if the vcpkg path in the vcpkg toolchain that the user has set is inconsistent with the vcpkg path set in the current extension?
2. If the user disables vcpkg in the extension, the vcpkg toolchain should be removed from cmake settings because this will affect the cmake configuration. At this point, should we inform the user or let the user make a choice?

Reply:
I think the problem here is multiple sources of truth. In my opinion, the user would generally prefer using the env variable since it encourages a cross-platform build system with no change. But you should decide on one source of truth. Either let the extension setting decide everything or use user-provided path in settings.json as a source of truth.

Reply:
I think we should warn user and give them 2 choise:
- Disable vcpkg.
- Override vcpkg toolchain in cmake settings using `vcpkg.general.vcpkgPath`

Reply:
Will this `$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake` match against an absolute path? Since you mentioned VScode settings have to use absolute path string.

Reply:
> Will this `$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake` match against an absolute path? Since you mentioned VScode settings have to use absolute path string.

The validation function will handle these environment variables, so they are equivalent to the outside function.